src/hooks/useMediaQuery.ts

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect, useCallback } from "react";
+
+const useMediaQuery = (minWidth: number): boolean => {
+  const query = `screen and (min-width: ${minWidth}px`;
+
+  const doesMediaMatches = useCallback((): boolean => {
+    if (typeof window === "undefined") {
+      return false;
+    }
+
+    return window.matchMedia(query).matches;
+  }, [query]);
+
+  const [matches, setMatches] = useState(doesMediaMatches());
+
+  const handleSetMatches = useCallback(() => {
+    setMatches(doesMediaMatches());
+  }, [doesMediaMatches]);
+
+  useEffect(() => {
+    handleSetMatches();
+
+    window.addEventListener("resize", handleSetMatches);
+
+    return () => {
+      window.removeEventListener("resize", handleSetMatches);
+    };
+  }, [handleSetMatches]);
+
+  return matches;
+};
+
+export { useMediaQuery };
```

Comment:
Tu usou para mostrar ou não certas imagens né ? legal a abordagem mas para imagens em vez de seguir essa abordagem é mais recomendado fazer usando css mesmo, acaba que tu pode trazer layout shift trocando as imagens dessa forma, talvez uma solução legal seria criar um componente que abstrai isso dos breakpoints e se vai mostrar ou não, mas pode fazer utilizando css, coloca as duas imagens no dom mas só mostra X, se breakpoint for X sabe

Reply:
Aqui eu usei isso para setar que imagem o Image do next iria usar no src, tentei com css tbm mas ele meio que não aceita então tive que "apelar" pra isso, saberia alguma boa forma de fazer usando o Image?